packages/filebrowser-extension/schema/widget.json

```diff
@@ -0,0 +1,82 @@
+{
+  "title": "File Browser Widget",
+  "description": "File Browser widget settings.",
+  "jupyter.lab.toolbars": {
+    "FileBrowser": [
```

Comment:
How does this name play with other file browser instances? For example created with the following code snippet:

```ts
const { createFileBrowser } = browser;
const widget = createFileBrowser('custom-file-browser', {
    driveName: 'custom-drive-name'
});
```

Reply:
Any `FileBrowser` instance will be created with an empty toolbar (as well as the default actually). The default widget toolbar is populated in a separated step in `browserWidget` plugin. When a developer creates a new file browser, it will be up to him to populate the toolbar (reusing or not the default file browser toolbar definition).

For example, if the new file browser should have the same toolbar, the following snippet will do:

```ts
const { createFileBrowser } = browser;
const widget = createFileBrowser('custom-file-browser', {
    driveName: 'custom-drive-name'
});

setToolbar(
  widget,
  createToolbarFactory(
    toolbarRegistry,
    settings,
    'FileBrowser',
    '@jupyterlab/filebrowser-extension:widget',
    translator
  )
);
```

If he wants to provide a list of toolbar widgets directly without using settings:

```ts
const { createFileBrowser } = browser;
const widget = createFileBrowser('custom-file-browser', {
    driveName: 'custom-drive-name'
});

setToolbar(
  widget,
  (browser: Widget) => [
    {
      "name": "item1",
      "widget": new ToolbarButton({...})
    },
    ...
  ]
);
```

Reply:
Ah nice thanks for the extra context.

Should we add a new section for this next to the other settings-based customization in the documentation?

https://jupyterlab.readthedocs.io/en/latest/extension/extension_points.html?#toolbar-registry